server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,635 @@
+                .isNull();
+    }
+
+
+    static Stream<Arguments> mailDirectiveShouldIgnoreBccHeadersParamsProvider() throws Exception {
+        return Stream.of(
+            Arguments.of(Rule.Condition.Comparator.CONTAINS, mimeMessageBuilder().addBccRecipient(RECIPIENT_TO_1_FULL_ADDRESS), RECIPIENT_TO_1_FULL_ADDRESS),
+            Arguments.of(Rule.Condition.Comparator.EXACTLY_EQUALS, mimeMessageBuilder().addBccRecipient(RECIPIENT_TO_1_FULL_ADDRESS), RECIPIENT_TO_1_FULL_ADDRESS)
+        );
+    }
+
+    @ParameterizedTest(name = "mailDirectiveShouldIgnoreBccHeaders with comparator {0}")
+    @MethodSource("mailDirectiveShouldIgnoreBccHeadersParamsProvider")
+    void mailDirectiveShouldIgnoreBccHeaders(
+            Rule.Condition.Comparator comparator,
+            MimeMessageBuilder mimeMessageBuilder,
+            String valueToMatch,
+            JMAPFilteringTestSystem testSystem) throws Exception {
+
+        testSystem.defineRuleForRecipient1(Rule.Condition.Field.RECIPIENT, comparator, valueToMatch);
+
+        FakeMail mail = FakeMail.builder()
+                .sender(SENDER_2_ADDRESS)
+                .recipients(RECIPIENT_1)
+                .mimeMessage(mimeMessageBuilder)
+                .build();
+
+        testSystem.getJmapFiltering().service(mail);
+
+        assertThat(mail.getAttribute(DELIVERY_PATH_PREFIX + RECIPIENT_1_USERNAME))
+                .isNull();
+    }
+
+
+
+
+    static Stream<Arguments> mailDirectiveShouldNotBeSetWhenHeaderContentIsNullParamsProvider() throws Exception {
+        return Stream.of(
+            Arguments.of(Rule.Condition.Field.FROM, Rule.Condition.Comparator.CONTAINS, mimeMessageBuilder(), SENDER_1_USERNAME),
+            Arguments.of(Rule.Condition.Field.FROM, Rule.Condition.Comparator.EXACTLY_EQUALS, mimeMessageBuilder(), SENDER_1_USERNAME),
+
+            Arguments.of(Rule.Condition.Field.TO, Rule.Condition.Comparator.CONTAINS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+            Arguments.of(Rule.Condition.Field.TO, Rule.Condition.Comparator.EXACTLY_EQUALS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+
+            Arguments.of(Rule.Condition.Field.CC, Rule.Condition.Comparator.CONTAINS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+            Arguments.of(Rule.Condition.Field.CC, Rule.Condition.Comparator.EXACTLY_EQUALS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+
+            Arguments.of(Rule.Condition.Field.RECIPIENT, Rule.Condition.Comparator.CONTAINS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+            Arguments.of(Rule.Condition.Field.RECIPIENT, Rule.Condition.Comparator.EXACTLY_EQUALS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+
+            Arguments.of(Rule.Condition.Field.SUBJECT, Rule.Condition.Comparator.CONTAINS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME),
+            Arguments.of(Rule.Condition.Field.SUBJECT, Rule.Condition.Comparator.EXACTLY_EQUALS, mimeMessageBuilder(), RECIPIENT_TO_1_USERNAME)
```

Comment:
The `mimeMessageBuilder()` & the `RECIPIENT_TO_1_USERNAME` arguments are alwyas the same: you can simplify it...